go/internal/ble-driver/BertyDevice_darwin.m

```diff
@@ -333,65 +379,90 @@ - (void)peripheral:(CBPeripheral *)peripheral didOpenL2CAPChannel:(CBL2CAPChanne
+    //                        [self.manager cancelPeripheralConnection:self.peripheral];
+    //                    }
+    //                }
+    //            } else {
+    //                os_log_debug(OS_LOG_BLE, "🟡 didUpdateValueForCharacteristic: peerID unknown in connectedPeers");
+    //                ConnectedPeer *peer = [[ConnectedPeer alloc] init];
+    //                [peer setClient:self];
+    //                [peer setClientReady:TRUE];
+    //                [PeerManager addPeer:peer forPeerID:remotePeerID];
+    //            }
+    //        });
+        } else {
+            os_log_debug(OS_LOG_BLE, "didUpdateValueForCharacteristic: peerID characteristic error: %@", error);
+        }
         
-        dispatch_async(dispatch_get_global_queue(0, 0), ^{
-            ConnectedPeer *peer = [PeerManager getPeer:remotePeerID];
-            if (peer) {
-                os_log_debug(OS_LOG_BLE, "🟢 didUpdateValueForCharacteristic: peerID known in connectedPeers");
-                if ([peer channel] == nil && self.manager.psm != 0) {
-                    @synchronized ([peer channel]) {
-                        os_log_debug(OS_LOG_BLE, "🟢 didUpdateValueForCharacteristic: identifier=%{public}@ opening l2cap channel", [[peripheral identifier] UUIDString]);
-                        self.l2capLatch = [[CountDownLatch alloc] init:1];
-                        [peripheral openL2CAPChannel:self.psm];
-                        os_log_debug(OS_LOG_BLE, "🟢 didUpdateValueForCharacteristic: identifier=%{public}@ before wait", [[peripheral identifier] UUIDString]);
-                        [self.l2capLatch await];
-                        os_log_debug(OS_LOG_BLE, "🟢 didUpdateValueForCharacteristic: identifier=%{public}@ after wait", [[peripheral identifier] UUIDString]);
-                        [self.l2capLatch release];
-                        self.l2capLatch = nil;
-
-                        [peer setClient:self];
-                        [peer setClientReady:TRUE];
-                        if (![self checkAndHandleFoundPeer]) {
-                            [self.manager cancelPeripheralConnection:self.peripheral];
-                        }
-                    }
-                } else {
-                    [peer setClient:self];
-                    [peer setClientReady:TRUE];
-                    if (![self checkAndHandleFoundPeer]) {
-                        [self.manager cancelPeripheralConnection:self.peripheral];
-                    }
-                }
-            } else {
-                os_log_debug(OS_LOG_BLE, "🟡 didUpdateValueForCharacteristic: peerID unknown in connectedPeers");
-                ConnectedPeer *peer = [[ConnectedPeer alloc] init];
-                [peer setClient:self];
-                [peer setClientReady:TRUE];
-                [PeerManager addPeer:peer forPeerID:remotePeerID];
-            }
-        });
+        [self.writeQ completedTask:error];
```

Comment:
If this line means "completedTask without error" can you change it for `[self.writeQ completedTask:nil];`.
It's very unclear that error is nil at this point.

Also, is it normal that the various `os_log_debug` that log an error (lines 457 and 464) do not call this callback with an error?